website_multi_company/models/res_users.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+from odoo.http import request
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def _signup_create_user(self, values):
+        values.update({
```

Comment:
Why is this needed? If this is not working on core, shouldn't this be a patch upstream?

Reply:
If there are already multiple websites existing in the db, this is used to make sure that they all have the managed user. Otherwise, we start in an unknown state.

(oops I put this on the wrong line first time sorry)

Reply:
I don't know if this could be considered as not working in core. The design of core is that a [template user](https://github.com/odoo/odoo/blob/6921da00b3a253ea5d0cdd135a8629ef9cb14ca7/addons/auth_signup/models/res_users.py#L81) is used to create the users that are created by `auth_user`. 

The fix in core would be to have one template user per website, but that would require a new module joining `auth_user` and `website`. In the instance of that, I would just submit the module to OCA so that it is guaranteed to stay Open Source.

I don't think the added logic is necessary at this time, although it would be a nice to have allowing for the auto-manage of both the public and template users at the website level. If we're not exposing the template user to the administrator for editing though, it doesn't seem like that much of an advantage over simply injecting the values.

Reply:
OK, please investigate further is this is really needed (this is one of the uncovered methods in tests), and if so, put a docstring explaining its usage.